apps/agent/app/src/components/Balances.js

```diff
@@ -10,40 +10,63 @@ const CONVERT_THROTTLE_TIME = 5000
 const convertApiUrl = symbols =>
   `${CONVERT_API_BASE}/price?fsym=USD&tsyms=${symbols.join(',')}`
 
-class Balances extends React.Component {
-  state = {
-    convertRates: {},
-  }
-  componentDidMount() {
-    this.updateConvertedRates(this.props)
-  }
-  componentWillReceiveProps(nextProps) {
-    this.updateConvertedRates(nextProps)
-  }
-  updateConvertedRates = throttle(async ({ balances }) => {
-    const verifiedSymbols = balances
-      .filter(({ verified }) => verified)
-      .map(({ symbol }) => symbol)
+const Balances = React.memo(function Balances({ balances }) {
+  const [convertRates, setConvertRates] = useState({})
+  const [balanceItems, setBalanceItems] = useState([])
+  const { layoutName } = useLayout
+  const compactMode = layoutName === 'small'
+  // Fetches the conversion rate for the verified tokens
+  const updateConvertedRates = useCallback(
+    throttle(async balances => {
+      const verifiedSymbols = balances
+        .filter(({ verified }) => verified)
+        .map(({ symbol }) => symbol)
 
-    if (!verifiedSymbols.length) {
-      return
-    }
+      if (!verifiedSymbols.length) {
+        return
+      }
+
+      const res = await fetch(convertApiUrl(verifiedSymbols))
+      const convertRates = await res.json()
+      setConvertRates(convertRates)
+    }, CONVERT_THROTTLE_TIME),
+    []
+  )
+
+  useEffect(() => {
+    let cancelled = false
+
+    const updateConvertedRates = throttle(async balances => {
```

Comment:
I don’t think `throttle()` would work here? Everytime the `useEffect()` gets refresh, it recreates `updateConvertedRates`.

Reply:
Aaah very true.